### DISCUSSION TITLE
2022 Summer Virtual Internship Program

### DISCUSSION THREAD

**Discussion by ChristopherA** [BlockchainCommons/Community#74] [General]
Blockchain Commons is repeating our popular blockchain Virtual Internship program this upcoming Summer. If you'd like to know more about blockchain technology, human-rights privacy & decentralized identity, or if you're considering a career in these emerging fields, we invite you to apply.

We are looking for interns with a wide range of expertise including not just software engineers and hardware developers, but also pre-law students, library science students, and technical writers.

For more details on our first Virtual Internship program see this [Coindesk Article](https://www.coindesk.com/tech/2020/09/23/blockchain-commons-internship-introduces-new-developers-to-open-source/).

## The Basics

* Application deadline is Friday April 22, 2022.
* Program is Monday May 23, 2022 to August 15, 2022. The first month is more project planning, the bulk of the work is done during the last 8 weeks to focus on the project deliverables (June 20th through August 15th).
* 40 hours of professional effort towards internship projects is required over that time (over and above educational opportunities, office hours, mentorship, etc.)
* Interns are also expected to actively participate in a private Signal group with fellow interns and members of the Blockchain Commons community, as well as weekly open "intern office hours".
* A small honorarium of at least US$564 in Bitcoin is offered for interns who meet three milestones.

## The Projects

The Blockchain Commons Internship is a collaborative environment where our interns work together on projects. We expect to mold these projects to fit the passions of interns, provided they also support [the vision](https://www.blockchaincommons.com/vision.html) of Blockchain Commons. The following are the kinds of projects we are currently considering (but we are open to other ideas):

* Expansion of our Gordian system, which uses torgaps & airgaps to create an architecture for Bitcoin usage that supports independence, privacy, resilience, and openness — putting users in control of their digital assets.
* Bundling of existing Blockchain Commons software such as Gordian into Debian packages that can be automatically installed onto security-focused systems like Tails OS.
* Research & proof-of-concepts for alternative architectures to improve the security & privacy of platforms requiring hot wallets like Lightning & Ethereum, based on lessons from Gordian approaches that have proven to work for Bitcoin.
* Creation of user documents to make those tools accessible to even non-technological users.
* Interviewing of activists to support the development of new use cases, engagement models( similar to Rebooting Web of Trust's [Amira](https://w3c-ccg.github.io/amira/)), to allow everyone to better understand the use cases for activism.
* Expanding and polishing previous internship projects such as [Spotbit](https://www.Spotbit.info) and [Mori](https://github.com/BlockchainCommons/mori-cli).
* Working on non-technical projects including support of legal and regulatory advocacy for human-rights focused digital identity for enabling blockchain technology and best practices.
* Marketing & promotion of Blockchain Commons projects and standards to the larger blockchain ecosystem (another great project for non-engineer interns).

We expect at least a third of our work in Summer 2022 to be focused on supporting activists. This has been a long-held goal at Blockchain Commons, where we're working to protect people, not just money. For the second year in a row, a grant from the [Human Rights Foundation](https://hrf.org) is putting this work front and center this summer. Many of the above projects that we are considering are purposefully focused on support for activists.

For more information on different internship project opportunities, see the discussion at https://github.com/BlockchainCommons/Community/discussions/76

## The Office Hours

Blockchain Commons isn't just about improving Blockchain Commons projects. It's also about helping to introduce the next generation of professionals to the blockchain ecosystem and open source. That means you.

So we'll introduce you to the technology and also to the people working in blockchain. That's the purpose of our weekly office hours: to plan for the work in our program, and also to allow you to talk with your fellow interns and to introduce you to special guest speakers from the industry.

## Other Programs

There are other Blockchain internship programs, so be sure that the Blockchain Commons program is right for you. 

We have great respect for both the [Summer of Bitcoin internship](https://www.summerofbitcoin.org/), who we hope to work with this year, and the [Chaincode Residencies](https://learning.chaincode.com/), which have been seminars the last few years. Both the Summer of Bitcoin and the original Chaincode Residencies are more intensive work programs that offer greater renumerance for full-time work; Blockchain Commons' internship is a lesser commitment in return for a smaller honorarium.

You should apply to this Virtual Internship program with Blockchain Commons instead, if:

* You are a self-starter who enjoys setting their own goals while simultaneously working with others in a collaborative environment;
* You want to get an understanding of how the secure open-source ecosystem works and how to meaningfully contribute to it;
* You want to investigate a broader scope of possibilities;
* You have expertise that may not be just pure engineering; or
* You prefer smaller projects with a smaller commitment.

## Initial Application

Email [Christopher Allen @ChristopherA](mailto:ChristopherA@LifeWithAlacrity.com)  with your CV, along with a cover letter that details highlights of some of the things you like to work on with Blockchain Commons, and what you’d like to get out of an internship with us. Please include what hardware platform, smart phone, and development environments you are most comfortable with, or what other expertises you have (such as pre-law or tech writing experience).

The deadline is Friday, April 22nd, 2022.

## Followup Questions

After initial application, you may be asked for additional details from the following list of questions: 

(Note that these questions are focused toward developers; we do not expect these skills for Law, Library Science, and Education interns, and would instead have different followup questions.)

* What is your primary development environment and what hardware do you run it on? What kind of smartphone do you use and which version of Android or iOS?

* As a software developer, what languages and tools do you LIKE, vs ones you know and can use.

* A key part of being successful as an intern is picking and scoping a new feature in an existing open-source project or doing so for a new small project that you will lead, that is relevant to Blockchain Commons and our advocacy and that can be completed in three months. You'll collaborate with the other interns with their projects, but there should be one where you are the lead. Given that, any early thoughts on what kinds of existing Blockchain Commons projects that might want to add a feature to? Or some small new project that fits into our advocacy?

* You clearly care about and have some advocacy around Bitcoin, blockchain, and related technologies, but what aspect of it drives you to do so?

* Software developers are whole people, not just coding machines. What professional skills and talents do you bring to the table that are not about code?

* What specifically would you like to learn or get out of a 3-month internship with Blockchain Commons? What appealed to you that you are applying for an internship with us?

* Look through our list of [potential internship projects](https://github.com/BlockchainCommons/Community/discussions/76). Are any of particular interest to you take take a lead role on? Any that you'd like to take a supporting role with? (We ask all interns to do one of each.)

## Corporate 

We are open to working with organizations to provide [“Externship”](https://en.wikipedia.org/wiki/Externship/) / ["Secondment"](https://en.wikipedia.org/wiki/Secondment) opportunities to help increase professional skills of their staff in open-source secure development best practices. For a fee we can also offer "pair programming"-like opportunities with specific professional open-source developers. Contact Christopher Allen for more information.

## Sponsoring our Internship Program

Thank you to [HRF](https://hrf.org/) for their 2nd grant for our virtual summer internship program!

To join in the support of our Virtual Internship Program, and other educational and career programs at Blockchain Commons, please consider becoming a [Sponsor](https://github.com/sponsors/BlockchainCommons) here at GitHub. However, you can alternatively support Blockchain Commons and our vision of the open web with Bitcoin via our [BTCPay contribution](https://btcpay.blockchaincommons.com) page.

We look forward to collaborating with you!

**Comment by Brisa-Codes** [comment]
Can’t miss this 🤝☺️

**Reply by MdRaiesh** [reply]
How to apply for this

**Reply by ChristopherA** [reply]
just email me ChristopherA@LifeWithAlacrity.com your details.

**Comment by ChristopherA** [comment]
An early start on some possible summer internship project ideas is at https://github.com/BlockchainCommons/Community/discussions/74

**Comment by anipaul2** [comment]
Interested

**Comment by SaySayo** [comment]
Interested

**Comment by SiddheshKukade** [comment]
Hey @ChristopherA are you also going to continue this program for 2023 ?
